packages/odyssey-react/src/components/PasswordInput/PasswordInput.test.tsx

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import { PasswordInput } from ".";
+
+const textBox = "textbox";
+const label = "Destination";
+
+describe("PasswordInput", () => {
+  beforeEach(() => {
+    jest
+      .spyOn(window, "requestAnimationFrame")
+      .mockImplementation((callback: FrameRequestCallback): number => {
+        callback(0);
+        return 0;
+      });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("renders into the document", () => {
+    render(<PasswordInput label={label} />);
+    expect(screen.getByLabelText(label)).toBeVisible();
+  });
+
+  it("has a button that changes the type when clicked", () => {
+    render(<PasswordInput label={label} defaultValue="Imma password" />);
+    const inputElement = screen.getByLabelText(label);
+    expect(inputElement).toHaveValue("Imma password");
+    const eyeButton = screen.getByRole("button");
+    expect(eyeButton).toBeInTheDocument();
+    eyeButton.click();
+    expect(screen.getByRole(textBox)).toBeInTheDocument();
```

Comment:
I think this part of the test is a little bit confusing because it would be true before the button click as well? If I'm tracking correctly we might use a different test setup 

We could consider a different assertion like `toHaveAttribute("type", "...")` to make the intent even more clear

Reply:
When `type=password` the role textbox is no longer exposed/reported. I'll update the test to be more clear by using the `toHaveAttribute`

Reply:
gotcha, or asserting against a role of `password` ? perhaps the aria behavior doesn't work that way would have to verify